mongodb/leafygreen-ui
# Popover upgrades

## ✍️ Proposed changes

This PR does 3 main things:
- Adds support for setting portal containers, including scrollable ones, to Popover and all components based on it through a consistent props interface.
- Improves the accessibility of the focus states for the side navigation.
- Introduces a prop for Popover and all components based on it to set z-indexes directly.

🎟 _Jira ticket:_ [Support zIndex prop for components using a Popover](https://jira.mongodb.org/browse/PD-729)
🎟 _Jira ticket:_ [Allow setting popover scroll containers](https://jira.mongodb.org/browse/PD-1278)

## 🛠 Types of changes




## ✅ Checklist




## 🧪 How to test changes

This list of ways to test this PR isn't comprehensive, but should cover *most* use cases of this new and fixed functionality.

Side navigation focus states:
- Open https://mongodb.design, and tab through the side navigation on the site. All focus states should be higher contrast with the background now.

`zIndex` prop:
- Open storybook.
- For each popover component (Popover, Tooltip, Menu, Select), verify that adjusting the z-index knob updates the z-index CSS property on the popover being rendered.

MongoDB.design popover fix, and scrollContainer provider validation:
- Open MongoDB.design, and expand one of the select knobs.
- When you scroll the page, you should see the select dropdown scroll with it.

Portal container prop:
- For each popover component, with usePortal set to `true`, pass a reference to an HTML element to the popoverContainer prop.
- You should see the popover component render within that container.

LeafyGreenProvider portal container setting:
- When rendering a component using a popover within the provider, set `popover.portalContainer` to an element reference. You should see any popovers within that provider that don't locally have the `portalContainer` prop set be rendered within that container.
